### PR TITLE
Task persistence fix: proper resource restore edge-case handling

### PR DESCRIPTION
**Pull request by mfranciszkiewicz**
- fixes an edge case of re-adding resources to HyperG
- encapsulates HyperG daemon's error responses in `HTTPError`s

### CODE REVIEW

File: golem/resource/hyperdrive/resourcesmanager.py [hunk 1]
```diff
@@ -94,7 +94,7 @@ def _add_files(self, files, task_id, resource_hash=None,
                                         id=task_id,
                                         client_options=client_options,
                                         obj_id=str(uuid.uuid4()),
-                                        raise_exc=bool(resource_hash))
+                                        raise_exc=True)
```
Comment (maaktweluit) [inline golem/resource/hyperdrive/resourcesmanager.py:97]: is this flag still used elsewhere? if all change to `True` it can become default behaviour

**Review by maaktweluit — APPROVED** [review]
LGTM

File: golem/resource/hyperdrive/resourcesmanager.py (hunk 1 above)
Comment (mfranciszkiewicz) [reply golem/resource/hyperdrive/resourcesmanager.py:97]: Yes, it is used in other places. BTW this part of resource management is going to be refactored.

**Comment by Elfoniok** [comment]
That solves problem discovered yesterday with some deserialized tasks crashing applications.

**Review by jiivan — APPROVED** [review]
LGTM

**Review by Elfoniok — APPROVED** [review]

**Pull request merged**